src/ecommerce.interfaces.ts

```diff
@@ -0,0 +1,160 @@
+        tax?: string | number
+    ): TransactionAttributes | null;
+    expandCommerceEvent(event: CommerceEvent): SDKEvent[] | null;
+}
+
+export interface SDKCart {
+    add(product: SDKProduct | SDKProduct[], logEvent?: boolean): void;
+    remove(product: SDKProduct | SDKProduct[], logEvent?: boolean): void;
+    clear(): void;
+}
+
+// Used for the public `eCommerce` namespace
+export interface SDKECommerceAPI extends IECommerceShared {
+    /*
+     * @deprecated
+     */
+    Cart: SDKCart;
+
+    logCheckout(
+        step: number,
+        option?: string,
+        attrs?: SDKEventAttrs,
+        customFlags?: SDKEventCustomFlags
+    ): void;
+    logImpression(
+        impression: SDKProductImpression,
+        attrs?: SDKEventAttrs,
+        customFlags?: SDKEventCustomFlags,
+        eventOptions?: SDKEventOptions
+    ): void;
+    logProductAction(
+        productActionType: valueof<typeof ProductActionType>,
+        product: SDKProduct | SDKProduct[],
+        attrs?: SDKEventAttrs,
+        customFlags?: SDKEventCustomFlags,
+        transactionAttributes?: TransactionAttributes,
+        eventOptions?: SDKEventOptions
+    ): void;
+    logPromotion(
+        type: valueof<typeof PromotionActionType>,
+        promotion: SDKPromotion,
+        attrs?: SDKEventAttrs,
+        customFlags?: SDKEventCustomFlags,
+        eventOptions?: SDKEventOptions
+    ): void;
+    logPurchase(
+        transactionAttributes: TransactionAttributes,
+        product: SDKProduct | SDKProduct[],
+        clearCart?: boolean,
+        attrs?: SDKEventAttrs,
+        customFlags?: SDKEventCustomFlags
+    ): void;
+    logRefund(
+        transactionAttributes: TransactionAttributes,
+        product: SDKProduct | SDKProduct[],
+        clearCart?: boolean,
+        attrs?: SDKEventAttrs,
+        customFlags?: SDKEventCustomFlags
+    ): void;
+    setCurrencyCode(code: string): void;
+}
+
+// Used for the private `_Ecommerce` namespace
+export interface IECommerce extends IECommerceShared {
+    buildProductList(event: any, product: Product | Product[]): Product[];
+    convertProductActionToEventType(
+        productActionType: valueof<typeof ProductActionType>
+    ): typeof CommerceEventType | typeof EventType | null;
+    convertPromotionActionToEventType(
+        promotionActionType: valueof<typeof PromotionActionType>
+    ): typeof CommerceEventType | null;
+    convertTransactionAttributesToProductAction(
+        transactionAttributes: TransactionAttributes,
+        productAction: ProductAction
+    ): void;
+    createCommerceEventObject(
+        customFlags: SDKEventCustomFlags,
+        options?: SDKEventOptions
+    ): SDKEvent | null;
+    expandProductAction(commerceEvent: CommerceEvent): SDKEvent[];
+    expandProductImpression(commerceEvent: CommerceEvent): SDKEvent[];
+    expandPromotionAction(commerceEvent: CommerceEvent): SDKEvent[];
+    extractActionAttributes(
+        attributes: Record<string, string | number>,
```

Comment:
Can you add an interfacte/type for these product attributes to be used here and in the other `extract` methods below?  Could be a new ticket also

Reply:
i also think the same comment above about `void`/mutation vs returning a new object can apply to the `extract` methods as well, so make a note of that in the ticket